elastic/kibana
# [Stateful sidenav] Add ML group menu in oblt & search

In this PR I've added the ML navigation group to the Observability ~and Search~ (EDIT: from comments below, I didn't add the ML group to the search solution nav) solution navigation.

## Note on implementation

In the linked issue, the request is to add the ML group as a top level item inside the footer. The underlying EUI component does not render a custom element to be render for top level item. The "panel opener" with its icon on the right is a custom element.

There are 3 alternative to fix this:

1. Move the ML group inside the main body as a last element
2. Move the "Other tools" top level accordion inside footer and put the ML group inside the accordion as last item
3. Modify the EUI to allow a custom element to be rendered as top level item

[EDIT]: As per the comment below I moved the ML group under the "management" top level group in the footer.

## Discuss

* In Observability, there is already an "AI & ML" section which contains some of the links inside the panel. What do we do in this case? Should we remove that group entirely and leave all the links in the panel?

<img width="547" alt="Screenshot 2024-08-19 at 10 58 55" src="https://github.com/user-attachments/assets/0dc75a46-257b-4837-87d7-0edd57d32b24">

* When we use panels we need a landing page. Currently I am using the security ML landing page in both observability and search. Are we OK to have a URL with "security" inside the observability navigation?

<img width="1711" alt="Screenshot 2024-08-19 at 10 59 33" src="https://github.com/user-attachments/assets/65a6b8a4-3f2f-4d8e-ab2d-aded4803be54">

cc @kevinsweet 

## How to test

* Add the following in your `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
```

* Create a space setting the solution view to observability or search
* Verify that the ML menu renders correctly

## Screenshots

<img width="826" alt="Screenshot 2024-08-20 at 10 53 49" src="https://github.com/user-attachments/assets/c27546bc-9e77-4287-bd66-f8d029b6c90f">
<img width="549" alt="Screenshot 2024-08-20 at 10 54 00" src="https://github.com/user-attachments/assets/e0f97598-6753-45d5-8c43-543174ac34fa">

<img width="295" alt="Screenshot 2024-09-04 at 09 36 21" src="https://github.com/user-attachments/assets/2d652c4c-5745-414e-bafb-90e55717d89d">


Fixes https://github.com/elastic/kibana-team/issues/1023
